Tensorflow-quantum exception fails with "make docs"

Issue Description
-----------------
There is an exception when Tensorflow-quantum is not installed, so that the tests of the documentation executors should not throw an error. However, when running such tests with `make docs` from root directory in bash, two errors occur, as reported below. This is not the [expected behavior](https://mitiq.readthedocs.io/en/stable/contributing_docs.html#requirements). 

If tf-quantum is installed, with the additional command
```
>>> pip install -r docs/requirements.txt
```
then the tests run fine. Since this is the case in CI, this is not picked up by github actions.

How to Reproduce
----------------

### Code Snippet

```bash
>>> pip install -e .
>>> pip install -r requirements.txt
>>> make docs
```

### Error Output

```
Document: guide/guide-executors
-------------------------------
/miniconda3/envs/mit4/lib/python3.8/site-packages/pyquil/api/_compiler.py:398: UserWarning: No quilc server running at tcp://127.0.0.1:5555. Compilation using quilc will not be available.
  warnings.warn(f"{e}. Compilation using quilc will not be available.")
**********************************************************************
File "guide/guide-executors.rst", line 1041, in default
Failed example:
    import numpy as np
    import sympy
    import tensorflow as tf
    # tensorflow-quantum 0.4.0 is unavailable on Windows
    try:
        import tensorflow_quantum as tfq
        tfq_exists = True
    except ImportError:
        tfq_exists = False
    from cirq import Circuit


    def stochastic_bit_flip_simulation(circ: Circuit, p: float, num_monte_carlo: int = 100) -> float:
        """
        Simulates a circuit with random bit flip (X(\pi)) errors
        Args:
            circ: The quantum program as a cirq object
            p: probability of an X(\pi) gate on each qubit after each gate in circ
            num_monte_carlo: number of random trajectories to average over
        Returns:
            The expectation value of the 0 state as a float
        """
        nM = len(circ.moments)
        nQ = len(circ.all_qubits())

        # Create array of symbolic variables and reshape to natural circuit parameterization
        h = sympy.symbols(''.join(['h_{0} '.format(i) for i in range(nM * nQ)]), positive=True)
        h_array = np.asarray(h).reshape((nQ, nM))

        # Symbolicly add X gates to the input circuit
        noisy_circuit = Circuit()
        for i, moment in enumerate(circ.moments):
            noisy_circuit.append(moment)
            for j, q in enumerate(circ.all_qubits()):
                noisy_circuit.append(cirq.rx(h_array[j, i]).on(q))

        # rotations will be pi w/ prob p, 0 w/ prob 1-p
        vals = [np.reshape((np.random.rand(nQ, nM) < p) * np.pi, (1, nQ * nM)) for _ in range(num_monte_carlo)]

        # needs to be a rank 2 tensor
        vals = np.squeeze(vals)
        if num_monte_carlo == 1:
            vals = [vals]

        # Instantiate tfq layer for computing state vector
        state = tfq.layers.State()

        # Execute monte carlo sim with symbolic values specified by vals
        out = state(noisy_circuit, symbol_names=h, symbol_values=vals).to_tensor()

        # Fancy way of computing and summing individual density operators, follwed by averaging
        dm = tf.tensordot(tf.transpose(out), tf.math.conj(out), axes=[[1], [0]]).numpy() / num_monte_carlo

        # return measurement of 0 state
        return np.real(dm[0, 0])
Exception raised:
    Traceback (most recent call last):
      File "/miniconda3/envs/mit4/lib/python3.8/doctest.py", line 1336, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest default[0]>", line 3, in <module>
        import tensorflow as tf
    ModuleNotFoundError: No module named 'tensorflow'
**********************************************************************
File "guide/guide-executors.rst", line 1157, in default
Failed example:
    if tfq_exists:
        import cirq
        from mitiq.benchmarks import randomized_benchmarking

        circ = randomized_benchmarking.rb_circuits(1, [20], 1)[0]

        # Need to make sure the qubits are cirq.GridQubit
        circ=circ.transform_qubits(lambda q: cirq.GridQubit.rect(1, 1)[0])

        out = stochastic_bit_flip_simulation(circ, 0.001)
        assert 0.5 < out < 1
Exception raised:
    Traceback (most recent call last):
      File "/miniconda3/envs/mit4/lib/python3.8/doctest.py", line 1336, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest default[0]>", line 1, in <module>
        if tfq_exists:
    NameError: name 'tfq_exists' is not defined
**********************************************************************
1 items had failures:
   2 of  19 in default
19 tests in 1 items.
17 passed and 2 failed.
***Test Failed*** 2 failures.

Document: guide/guide-zne
-------------------------
2 items passed all tests:
  12 tests in default
  24 tests in python
36 tests in 2 items.
36 passed and 0 failed.
Test passed.
1 items passed all tests:
   1 tests in default (cleanup code)
1 tests in 1 items.
1 passed and 0 failed.
Test passed.

Document: guide/guide-hamiltonians
----------------------------------
<doctest python[0]>:11: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.
  plt.show()
1 items passed all tests:
  10 tests in python
10 tests in 1 items.
10 passed and 0 failed.
Test passed.

Document: guide/guide-ibmq-backends
-----------------------------------

Document: apidoc
----------------
1 items passed all tests:
   4 tests in default
4 tests in 1 items.
4 passed and 0 failed.
Test passed.

Doctest summary
===============
   87 tests
    2 failures in tests
    0 failures in setup code
    0 failures in cleanup code
build finished with problems, 2 warnings.
make[1]: *** [doctest] Error 1
make: *** [docs] Error 2
```

Environment Context
-------------------

Use the `about()` function to summarize information on operating system, python version and dependencies.
```python
Mitiq: A Python toolkit for implementing error mitigation on quantum computers
==============================================================================
Authored by: Mitiq team, 2020 & later (https://github.com/unitaryfund/mitiq)

Mitiq Version:	0.4.0dev

Core Dependencies
-----------------
Cirq Version:	0.9.1
NumPy Version:	1.18.5
SciPy Version:	1.4.1

Optional Dependencies
---------------------
PyQuil Version:	2.18.0
Qiskit Version:	0.12.0

Python Version:	3.8.5
Platform Info:	Darwin (x86_64)
```

--- a/docs/source/guide/guide-executors.rst
+++ b/docs/source/guide/guide-executors.rst
@@ -91,7 +91,7 @@ documentation `here <https://cirq.readthedocs.io/en/master/generated/cirq.PauliS
             obs: The observable to measure as a cirq.PauliString.
             shots: The number of measurements.
 
-        Returns: 
+        Returns:
             The expectation value of obs as a float.
         """
 
@@ -182,7 +182,7 @@ You can also include both noise models and finite sampling in your executor.
         """
         # add the noise
         noisy = circ.with_noise(depolarize(p=noise))
-        
+
         # Do the sampling
         psum = cirq.PauliSumCollector(noisy, obs, samples_per_term=shots)
         psum.collect(sampler=cirq.DensityMatrixSimulator())
@@ -355,7 +355,7 @@ behind how this example is available `here <https://quantumcomputing.stackexchan
             obs: The observable to measure as a NumPy array.
             shots: The number of measurements.
 
-        Returns: 
+        Returns:
             The expectation value of obs as a float.
 
         """
@@ -436,7 +436,7 @@ This executor can be used for noisy depolarizing simulation.
             noise: The depolarizing noise strength as a float, i.e. 0.001 is 0.1%.
             shots: The number of measurements.
 
-        Returns: 
+        Returns:
             The expectation value of obs as a float.
         """
         if len(circ.clbits) > 0:
@@ -522,9 +522,9 @@ Below is an example to use TensorFlow Quantum to simulate a bit-flip channel:
 
     import numpy as np
     import sympy
-    import tensorflow as tf
     # tensorflow-quantum 0.4.0 is unavailable on Windows
     try:
+        import tensorflow as tf
         import tensorflow_quantum as tfq
         tfq_exists = True
     except ImportError:
@@ -544,7 +544,7 @@ Below is an example to use TensorFlow Quantum to simulate a bit-flip channel:
         """
         nM = len(circ.moments)
         nQ = len(circ.all_qubits())
-    
+
         # Create array of symbolic variables and reshape to natural circuit parameterization
         h = sympy.symbols(''.join(['h_{0} '.format(i) for i in range(nM * nQ)]), positive=True)
         h_array = np.asarray(h).reshape((nQ, nM))
@@ -555,24 +555,24 @@ Below is an example to use TensorFlow Quantum to simulate a bit-flip channel:
             noisy_circuit.append(moment)
             for j, q in enumerate(circ.all_qubits()):
                 noisy_circuit.append(cirq.rx(h_array[j, i]).on(q))
-        
+
         # rotations will be pi w/ prob p, 0 w/ prob 1-p
         vals = [np.reshape((np.random.rand(nQ, nM) < p) * np.pi, (1, nQ * nM)) for _ in range(num_monte_carlo)]
-        
+
         # needs to be a rank 2 tensor
         vals = np.squeeze(vals)
         if num_monte_carlo == 1:
             vals = [vals]
-        
+
         # Instantiate tfq layer for computing state vector
         state = tfq.layers.State()
-        
+
         # Execute monte carlo sim with symbolic values specified by vals
         out = state(noisy_circuit, symbol_names=h, symbol_values=vals).to_tensor()
-        
+
         # Fancy way of computing and summing individual density operators, follwed by averaging
         dm = tf.tensordot(tf.transpose(out), tf.math.conj(out), axes=[[1], [0]]).numpy() / num_monte_carlo
-        
+
         # return measurement of 0 state
         return np.real(dm[0, 0])
 

